tests/test_zipapp.py

```diff
@@ -0,0 +1,35 @@
+import os.path
+import subprocess
+import sys
+
+import pytest
+
+import virtualenv
+
+
+def _python(v):
+    return virtualenv.get_installed_pythons().get(v, "python{}".format(v))
+
+
+@pytest.fixture(scope="session")
+def zipapp(tmp_path_factory):
+    if sys.version_info[:2] == (3, 4):
+        pytest.skip("zipapp was introduced in python3.5")
+    pyz = str(tmp_path_factory.mktemp(basename="zipapp") / "virtualenv.pyz")
+    subprocess.check_call(
+        (_python("3"), os.path.join(virtualenv.HERE, "tasks/make_zipapp.py"), "--root", virtualenv.HERE, "--dest", pyz)
```

Comment:
it was either tie the test to `virtualenv.py` location, the process `cwd`, or the test location -- I picked what I thought was the least bad -- do you have a better suggestion?

`FILE_PATH` is inaccurate, it's not actually the path to a file, but the path to the directory -- conventionally I've only seen this path referred to as `HERE`

Reply:
also can't use `sys.executable` in python2 because of `zipapp` -- that's why it's `_python("3")`

Reply:
this call is `tasks/make_zipapp.py`, not the call of the zipapp itself

Reply:
Oh, sorry. What you mean is if you're running virtualenv with Python 2, you need to use Python 3 to run make_zipapp? That makes sense, I guess. So if the user doesn't have Python 3 installed, this will fail. Not a real problem, as this is just a test.

Reply:
yep!

it's also the reason I had to skip python3.4 since `python3` in that case is too old to run `zipapp`

that end-of-life is coming up soon I think 🎉